frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -207,96 +247,129 @@ const ClusterServiceVersionStatus: React.FC<ClusterServiceVersionStatusProps> =
   ) : null;
 };
 
-export const ClusterServiceVersionTableRow = withFallback<ClusterServiceVersionTableRowProps>(
-  ({ obj, rowKey, subscription, catalogSourceMissing, index, style }) => {
-    const { displayName, provider, version } = _.get(obj, 'spec');
-    const [icon] = _.get(obj, 'spec.icon', []);
-    const deploymentName = _.get(obj, 'spec.install.spec.deployments[0].name');
-    const namespace = getNamespace(obj);
-    const route = resourceObjPath(obj, referenceFor(obj));
-    const uid = getUID(obj);
-    const internalObjects = getInternalObjects(obj);
-    return (
-      <TableRow id={uid} trKey={rowKey} index={index} style={style}>
-        {/* Name */}
-        <TableData className={tableColumnClasses[0]}>
-          <Link
-            to={route}
-            className="co-clusterserviceversion-link"
-            data-test-operator-row={displayName}
+export const NamespacedClusterServiceVersionTableRow = withFallback<
+  ClusterServiceVersionTableRowProps
+>(({ activeNamespace, obj, rowKey, subscription, catalogSourceMissing, index, style }) => {
+  const { displayName, provider, version } = _.get(obj, 'spec');
+  const {
+    'olm.targetNamespaces': olmTargetNamespaces = '',
+    'olm.operatorNamespace': olmOperatorNamespace = '',
+  } = obj.metadata?.annotations || {};
+  const [icon] = _.get(obj, 'spec.icon', []);
+  const route = resourceObjPath(obj, referenceFor(obj));
+  const uid = getUID(obj);
+  const internalObjects = getInternalObjects(obj);
+  const managedNamespaces = olmTargetNamespaces?.split(',') || [];
+  const managedNamespaceData = () => {
```

Comment:
This would be better as a separate React component instead of a function that returns markup.